hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/customize/ResourceServerConfigurationCustomizerDsl.kt

```diff
@@ -57,39 +67,58 @@ class ResourceServerConfigurationCustomizerBuilder : ResourceServerConfiguration
   private var authorizeHttpRequestsCustomizer = AuthorizeHttpRequestsCustomizerBuilder().build()
+  private var authorizeExchangeCustomizer = AuthorizeExchangeCustomizerBuilder().build()
   private var overrideAuthorizeHttpRequests = false
-  private var customizeAuthorizeHttpRequests = false
+  private var overrideAuthorizeExchange = false
+  private var customizeAuthorization = false
 
   override fun unauthorizedRequestPaths(dsl: UnauthorizedRequestPathCustomizerDsl.() -> Unit) =
     UnauthorizedRequestPathsCustomizerBuilder()
       .apply(dsl)
       .build()
       .also { unauthorizedRequestPathsCustomizer = it }
-      .also { customizeAuthorizeHttpRequests = true }
+      .also { customizeAuthorization = true }
 
   override fun anyRequestRole(dsl: AnyRequestRoleCustomizerDsl.() -> Unit): AnyRequestRoleCustomizer =
     AnyRequestRoleCustomizerBuilder()
       .apply(dsl)
       .build()
       .also { anyRequestRoleCustomizer = it }
-      .also { customizeAuthorizeHttpRequests = true }
+      .also { customizeAuthorization = true }
 
   override fun authorizeHttpRequests(dsl: AuthorizeHttpRequestsDsl.() -> Unit): AuthorizeHttpRequestsCustomizer =
     AuthorizeHttpRequestsCustomizerBuilder(dsl)
       .build()
       .also { authorizeHttpRequestsCustomizer = it }
       .also { overrideAuthorizeHttpRequests = true }
 
+  override fun authorizeExchange(dsl: AuthorizeExchangeDsl.() -> Unit): AuthorizeExchangeCustomizer =
+    AuthorizeExchangeCustomizerBuilder(dsl)
+      .build()
+      .also { authorizeExchangeCustomizer = it }
+      .also { overrideAuthorizeExchange = true }
+
   fun build(): ResourceServerConfigurationCustomizer {
-    if (overrideAuthorizeHttpRequests && customizeAuthorizeHttpRequests) {
-      throw IllegalStateException("Cannot override the entire authorizeHttpRequests DSL and try to customize it at the same time.")
-    }
+    validate()
 
     return ResourceServerConfigurationCustomizer()
       .also {
         it.unauthorizedRequestPathsCustomizer = unauthorizedRequestPathsCustomizer
         it.anyRequestRoleCustomizer = anyRequestRoleCustomizer
         it.authorizeHttpRequestsCustomizer = authorizeHttpRequestsCustomizer
+        it.authorizeExchangeCustomizer = authorizeExchangeCustomizer
       }
   }
+
+  private fun validate() {
+    if (overrideAuthorizeHttpRequests && customizeAuthorization) {
+      throw IllegalStateException("Cannot override the entire authorizeHttpRequests DSL and try to customize it at the same time.")
+    }
+    if (overrideAuthorizeExchange && customizeAuthorization) {
+      throw IllegalStateException("Cannot override the entire authorizeExchange DSL and try to customize it at the same time.")
+    }
+    if (overrideAuthorizeHttpRequests && overrideAuthorizeExchange) {
+      throw IllegalStateException("Cannot override both the authorizeHttpRequests DSL and authorizeExchange DSL as a Spring application cannot be both Servlet and Reactive based.")
+    }
+  }
```

Comment:
Add validation to prevent both reactive and non-reactive overrides at the same time.